cupaloy.go

```diff
@@ -66,6 +62,9 @@ func (c *config) SnapshotMulti(snapshotID string, i ...interface{}) error {
 
 func (c *config) SnapshotT(t *testing.T, i ...interface{}) {
 	t.Helper()
+	if !t.Failed() {
```

Comment:
This should just be if t.Failed() surely?

Unfortunately testing.T cannot be mocked so I'm unable to test these methods in CI yet